x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,502 @@
+    defaultMessage: 'Granularity',
+  });
+
+  return (
+    <>
+      <EuiFormRow
+        label={sectionLabel}
+        data-test-subj="indexPattern-ranges-section-label"
+        labelAppend={
+          <EuiSwitch
+            label={i18n.translate('xpack.lens.indexPattern.ranges.autoInterval', {
+              defaultMessage: 'Auto interval',
+            })}
+            checked={autoIntervalEnabled}
+            onChange={(e) => toggleAutoInterval(e.target.checked)}
+            data-test-subj="indexPattern-ranges-auto-interval"
+            compressed
+          />
+        }
+      >
+        <>
+          <EuiFlexGroup gutterSize="s" responsive={false} wrap>
+            <EuiFlexItem>
+              {autoIntervalEnabled ? (
+                <EuiRange
+                  min={1}
+                  max={HISTOGRAM_MAX_BARS}
+                  step={1}
+                  value={maxBars === 'auto' ? '' : maxBars}
+                  onChange={({ target }) =>
+                    onMaxBarsChange(Number(get(target, 'value', HISTOGRAM_MAX_BARS)))
+                  }
+                  placeholder={i18n.translate('xpack.lens.indexPattern.ranges.autoIntervals', {
+                    defaultMessage: 'Auto',
+                  })}
+                  showLabels
+                  showInput="inputWithPopover"
+                  prepend={i18n.translate('xpack.lens.indexPattern.ranges.maxIntervals', {
+                    defaultMessage: 'Max intervals',
+                  })}
+                />
+              ) : (
+                <EuiFieldNumber
+                  data-test-subj="lens-range-interval-field"
+                  value={interval}
+                  onChange={({ target }) => onIntervalChange(Number(target.value))}
+                  prepend={i18n.translate('xpack.lens.indexPattern.ranges.min', {
+                    defaultMessage: 'Min Interval',
+                  })}
+                  min={0}
+                />
+              )}
+            </EuiFlexItem>
+          </EuiFlexGroup>
+
+          <EuiButtonEmpty iconType="controlsHorizontal" onClick={() => onToggleEditor()}>
+            {i18n.translate('xpack.lens.indexPattern.ranges.customIntervalsToggle', {
+              defaultMessage: 'Create custom intervals',
+            })}
+          </EuiButtonEmpty>
+        </>
+      </EuiFormRow>
+    </>
+  );
+};
+
+const RangePopover = ({
+  range,
+  setRange,
+  Button,
+  isOpenByCreation,
+  setIsOpenByCreation,
+}: {
+  range: LocalRangeType;
+  setRange: (newRange: LocalRangeType) => void;
+  Button: React.FunctionComponent<{ onClick: MouseEventHandler }>;
+  isOpenByCreation: boolean;
+  setIsOpenByCreation: (open: boolean) => void;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [tempRange, setTempRange] = useState(range);
+
+  const setPopoverOpen = (isOpen: boolean) => {
+    setIsPopoverOpen(isOpen);
+    setIsOpenByCreation(isOpen);
+  };
```

Comment:
The popover behavior here looks buggy, I was seeing multiple open at once

<img width="514" alt="Screen Shot 2020-09-08 at 5 46 01 PM" src="https://user-images.githubusercontent.com/666475/92531887-29e33f00-f1fd-11ea-9119-af7ac7db2675.png">